### PR TITLE
Add C2B URL registration

### PR DESCRIPTION
**Pull request by weezqyd**
Implement C2B URL Registration API

### CODE REVIEW

File: mpesa_test.go [hunk 1]
```diff
@@ -650,3 +651,116 @@ func TestMpesa_STKPushQuery(t *testing.T) {
...
+		env        Environment
+		mock       func(t *testing.T, ctx context.Context, app *Mpesa, c *mockHttpClient, c2bRequest RegisterC2BURLRequest)
+		c2bRequest RegisterC2BURLRequest
+	}{
+		{
+			name: "it should register URLs in sanbox",
+			env:  Sandbox,
+			c2bRequest: RegisterC2BURLRequest{
+				ShortCode:       600638,
+				ResponseType:    "Complete",
```
Comment (jwambugu) [inline mpesa_test.go:668]: @weezqyd the tests are failing here
```suggestion
				ResponseType:    "Completed",
```

File: mpesa_test.go [hunk 2]
```diff
@@ -650,3 +651,116 @@ func TestMpesa_STKPushQuery(t *testing.T) {
...
+			mock: func(t *testing.T, ctx context.Context, app *Mpesa, c *mockHttpClient, c2bRequest RegisterC2BURLRequest) {
+				c.MockRequest(app.c2bURL, func() (status int, body string) {
+					req := c.requests[1]
+
+					asserts.Equal("application/json", req.Header.Get("Content-Type"))
+					wantAuthorizationHeader := `Bearer ` + app.cache[testConsumerKey].AccessToken
+					asserts.Equal(wantAuthorizationHeader, req.Header.Get("Authorization"))
+
+					var reqParams RegisterC2BURLRequest
+					//log.Println(fmt.Sprint(req.Body))
```
Comment (jwambugu) [inline mpesa_test.go:717]:
```suggestion
					
```

File: mpesa_test.go [hunk 3]
```diff
@@ -650,3 +651,116 @@ func TestMpesa_STKPushQuery(t *testing.T) {
...
```
Comment (jwambugu) [inline mpesa_test.go:681]:
```suggestion
					
```

File: mpesa.go [hunk 4]
```diff
@@ -342,3 +349,32 @@ func (m *Mpesa) STKQuery(
...
+
+		var result RegisterC2BURLResponse
+		err = json.NewDecoder(response.Body).Decode(&result)
+		if err != nil {
+			return nil, errors.Join(errors.New("mpesa: could not unmarshall c2b response body"), err)
+		}
+
+		return &result, nil
+	default:
+		return nil, fmt.Errorf("mpesa: the provided ResponseType [%s] is not valid.", req.ResponseType)
```
Comment (jwambugu) [inline mpesa.go:378]:
```suggestion
		return nil, fmt.Errorf("mpesa: the provided ResponseType [%s] is not valid", req.ResponseType)
```

**Review by jwambugu — CHANGES_REQUESTED** [review]

**Comment by weezqyd** [comment]
Hi @jwambugu 
I added the suggested fixes.

**Review by jwambugu — APPROVED** [review]

**Comment by jwambugu** [comment]
Merged @weezqyd

**Pull request merged**